flyteplugins/go/tasks/pluginmachinery/flytek8s/pod_helper.go

```diff
@@ -554,6 +556,31 @@ func BuildIdentityPod() *v1.Pod {
 	}
 }
 
+type PhaseInfoFailureWithCleanup struct {
+    PhaseVersion pluginsCore.PhaseVersion
+    Reason       string
+    Message      string
+    TaskInfo     *pluginsCore.TaskInfo
+    CleanupInfo  *CleanupInfo
+}
+
+// CleanupInfo represents information for cleanup tasks.
+type CleanupInfo struct {
+    CleanupReason   string
+    CleanupMessage  string
+}
+
+// NewPhaseInfoFailureWithCleanup creates a new PhaseInfoFailureWithCleanup instance.
+func NewPhaseInfoFailureWithCleanup(reason, message string, taskInfo *pluginsCore.TaskInfo, cleanupInfo *CleanupInfo) *PhaseInfoFailureWithCleanup {
+    return &PhaseInfoFailureWithCleanup{
+        PhaseVersion: pluginsCore.DefaultPhaseVersion,
+        Reason:       reason,
+        Message:      message,
+        TaskInfo:     taskInfo,
+        CleanupInfo:  cleanupInfo,
+    }
+}
+
```

Comment:
Hi, just a reminder, if the function is added, the test should be added too, thanks a lot!

Reply:
Done